packages/easy-coding-standard/src/HttpKernel/EasyCodingStandardKernel.php

```diff
@@ -4,84 +4,108 @@
+        $containerBuilder->registerExtension(new ConsoleColorDiffExtension());
+        $containerBuilder->registerExtension(new SkipperExtension());
+        $containerBuilder->registerExtension(new SymplifyCodingStandardExtension());
 
-        if (StaticVersionResolver::PACKAGE_VERSION !== '@package_version@') {
-            $logDirectory .= '_' . StaticVersionResolver::PACKAGE_VERSION;
-        }
+        // cleanup
+        $containerBuilder->addCompilerPass(new RemoveExcludedCheckersCompilerPass());
+        $containerBuilder->addCompilerPass(new RemoveMutualCheckersCompilerPass());
 
-        return $logDirectory;
-    }
+        // autowire
+        $containerBuilder->addCompilerPass(new AutowireInterfacesCompilerPass([
+            FixerInterface::class,
+            Sniff::class,
+            OutputFormatterInterface::class,
+        ]));
 
-    public function boot(): void
-    {
-        $cacheDir = $this->getCacheDir();
-
-        try {
-            FileSystem::delete($cacheDir);
-            FileSystem::createDir($cacheDir);
-        } catch (Throwable) {
-            // the "@" is required for parallel run to avoid deleting locked directory
-            // Rebuild the container on each run
+        // exceptions
+        $containerBuilder->addCompilerPass(new ConflictingCheckersCompilerPass());
+
+        // method calls
+        $containerBuilder->addCompilerPass(new FixerWhitespaceConfigCompilerPass());
+
+        // add compiler passes
+        $containerBuilder->addCompilerPass(new AutowireArrayParameterCompilerPass());
+
+        // load manually passed configs
+        $delegatingLoader = $this->createConfigLoader($containerBuilder);
+        foreach ($configFiles as $configFile) {
+            $delegatingLoader->load($configFile);
         }
 
-        parent::boot();
+        $extensionNames = array_map(
+            fn (ExtensionInterface $extension): string => $extension->getAlias(),
+            $containerBuilder ->getExtensions()
+        );
+
+        // this calls load() method in every extensions
+        // ensure these extensions are implicitly loaded
+        // @todo remove dependency on HttpKernel and create own light implementation with reference to original code
+        $compilerPassConfig = $containerBuilder->getCompilerPassConfig();
+        $compilerPassConfig->setMergePass(
+            new MergeExtensionConfigurationPass($extensionNames)
+        );
+
+        $containerBuilder->compile();
+        $this->container = $containerBuilder;
+
+        return $containerBuilder;
     }
 
-    protected function prepareContainer(ContainerBuilder $containerBuilder): void
+    public function getContainer(): ContainerInterface
     {
-        // works better with workers - see https://github.com/symfony/symfony/pull/32581
-        $containerBuilder->setParameter('container.dumper.inline_factories', true);
-        parent::prepareContainer($containerBuilder);
+        if ($this->container === null) {
```

Comment:
I think it can use instanceof check:

```suggestion
        if (! $this->container instanceof ContainerBuilder) {
```